src/test/storage/chunk_test.cpp

```diff
@@ -218,10 +218,21 @@ TEST_F(StorageChunkTest, RemoveIndex) {
 }
 
 TEST_F(StorageChunkTest, OrderedBy) {
-  EXPECT_EQ(chunk->ordered_by(), std::nullopt);
-  const auto ordered_by = std::make_pair(ColumnID(0), OrderByMode::Ascending);
-  chunk->set_ordered_by(ordered_by);
-  EXPECT_EQ(chunk->ordered_by(), ordered_by);
+  EXPECT_EQ(chunk->sorted_by(), std::nullopt);
+  auto sorted_by_vector = std::vector<SortColumnDefinition>{SortColumnDefinition(ColumnID{0}, SortMode::Ascending)};
+  chunk->finalize();
+  chunk->set_sorted_by(sorted_by_vector);
+  EXPECT_EQ(chunk->sorted_by(), sorted_by_vector);
+
+  sorted_by_vector = {SortColumnDefinition(ColumnID{0}, SortMode::Ascending),
+                      SortColumnDefinition(ColumnID{1}, SortMode::Descending)};
+  chunk->set_sorted_by(sorted_by_vector);
```

Comment:
What if an asynchronous process tries to find sort orders for chunks of data tables? In this case, we still allow to modify/extend the sort order.

Reply:
I was more thinking of this as an additional safety guard. If we actually end up having to call it twice, we can still remove that assert again.

Reply:
Something like `Assert(new.size() > 1 && old.size() == 0, "");`. Currently, I tend to think that removing the sorted information signals that something is wrong and should thus never happen. If we add a new vector, the current one needs to be empty.

Does that make sense to you?

Reply:
Exactly that.

Reply:
```
// For now, we assume that resetting (either to no-sorting or a new sorting) of chunks signals that something went
  // wrong (chunks cannot be sorted in-place). This might change when, e.g., background processes try to determine
  // columns which are "accidentally" sorted and add this information later on.
  Assert(sorted_by.size() > 1 && _sorted_by.size() == 0, "Sorting information cannot be empty or resetted.");.
```

Ok?

Reply:
Currently, we assume that set_sorted_by is called only once at most. As such, there should be no existing sorting and the new sorting should contain at least one column. Feel free to remove this assertion if necessary.